docs/source/developer_guide/core/internals.rst

```diff
@@ -15,10 +15,14 @@ This means that as soon as a node is stored any attempt to alter its attributes,
 Certain subclasses of nodes need to adapt this behavior however, as for example in the case of the :py:class:`~aiida.orm.node.process.process.ProcessNode` class (see `calculation updatable attributes`_), but since the immutability
 of stored nodes is a core concept of AiiDA, this behavior is nonetheless enforced on the node level. This guarantees that any subclasses of the Node class will respect this behavior unless it is explicitly overriden.
 
+Node methods
+******************
+- :py:meth:`~aiida.orm.implementation.general.node.clean_value` takes a value and returns an object which can be serialized for storage in the database. Such an object must be able to be subsequently deserialized without changing value. If a simple datatype is passed (integer, float, etc.), a check is performed to see if it has a value of ``nan`` or ``inf``, as these cannot be stored. Otherwise, if a list, tuple, dictionary, etc., is  passed, this check is performed for each value it contains. This is done recursively, automatically handling the case of nested objects. It is important to note that iterable type objects are converted to lists during this process, and mappings, such as dictionaries, are converted to normal dictionaries. This cleaning process is used by default when setting node attributes via :py:meth:`~aiida.orm.implementation.general.node.AbstractNode._set_attr` and :py:meth:`~aiida.orm.implementation.general.node.AbstractNode._append_to_attr`, although it can be disabled by setting ``clean=False``. Values are also cleaned when setting extras on a stored node using :py:meth:`~aiida.orm.implementation.general.node.AbstractNode.set_extras` or :py:meth:`~aiida.orm.implementation.general.node.AbstractNode.reset_extras`, but this cannot be disabled. 
```

Comment:
A couple of small notes here:
- I don't think `NaN` and `inf` are already checked - this is part of #1498 (for me it's fine if you want to solve both issues in this PR but then you need to add the logic to check for these values).
- maybe a short mention that at `.store()` clean_value is not called, so if you use `clean=False` you should be very sure that you are only using base types (e.g. not tuples but lists etc.) otherwise e.g. the hashing will be wrong, and depending on the DB backend you can either get an exception while storing (I guess in the case of numpy arrays), or the type can be converted silently (I guess in the case of tuples).

Reply:
Hi Giovanni, looking again at this, it's looks like clean_builtin() does this check:
https://github.com/aiidateam/aiida_core/blob/7929d2333c3811c9ef5c520c765f32eac71e3b0c/aiida/orm/implementation/general/node.py#L61

Reply:
Thanks, you are right! I didn't realise this was already fixed in #2129 but we didn't close #1498